controllers/api/imap.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+
+	ctx "github.com/gophish/gophish/context"
+	"github.com/gophish/gophish/imap"
+	log "github.com/gophish/gophish/logger"
+	"github.com/gophish/gophish/models"
+)
+
+// IMAPServerValidate handles requests for the /api/imapserver/validate endpoint
+func (as *Server) IMAPServerValidate(w http.ResponseWriter, r *http.Request) {
+	switch {
+	case r.Method == "GET":
+		JSONResponse(w, models.Response{Success: false, Message: "Only POSTs allowed"}, http.StatusBadRequest)
+	case r.Method == "POST":
+		im := models.IMAP{}
+		err := json.NewDecoder(r.Body).Decode(&im)
+		if err != nil {
+			JSONResponse(w, models.Response{Success: false, Message: "Invalid request"}, http.StatusBadRequest)
+			return
+		}
+		err = imap.Validate(&im)
+		if err != nil {
+			JSONResponse(w, models.Response{Success: false, Message: err.Error()}, http.StatusOK)
+			return
+		}
+		JSONResponse(w, models.Response{Success: true, Message: "Successful login."}, http.StatusCreated)
+	}
+}
+
+// IMAPServer handles requests for the /api/imapserver/ endpoint
+func (as *Server) IMAPServer(w http.ResponseWriter, r *http.Request) {
+	switch {
+	case r.Method == "GET":
+		ss, err := models.GetIMAP(ctx.Get(r, "user_id").(int64))
+		if err != nil {
+			log.Error(err)
```

Comment:
We should probably send back a `JSONResponse` with the error here and the status for an internal server error, then return.

Reply:
Good point, not sure why I didn't. Fixed.